<?xml version="1.0" encoding="utf-8"?>

<launch>

    <!-- # args -->
    <arg name="use_map" default="true" />

    <!-- # move_base -->
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">

        <rosparam file="$(find ros_my_move_base)/config/basic.yaml" command="load" />

        <!-- # planner global -->
        <rosparam file="$(find ros_my_move_base)/config/planner/GlobalPlanner.yaml" command="load" />

        <!-- # planner local -->
        <rosparam file="$(find docking_sim)/config/dock/planner/DWAPlannerROS.yaml" command="load" />

        <!-- # costmap common -->
        <rosparam file="$(find docking_sim)/config/dock/costmap/common.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find docking_sim)/config/dock/costmap/common.yaml" command="load" ns="local_costmap" />

        <!-- # costmap global -->
        <rosparam if="$(arg use_map)" file="$(find ros_my_move_base)/config/costmap/global_use_map.yaml" command="load" ns="global_costmap" />
        <rosparam unless="$(arg use_map)" file="$(find ros_my_move_base)/config/costmap/global.yaml" command="load" ns="global_costmap" />

        <!-- # costmap local -->
        <rosparam file="$(find ros_my_move_base)/config/costmap/local.yaml" command="load" ns="local_costmap" />

        <!-- # remap -->
        <remap from="cmd_vel" to="cmd_vel_car1" />
    </node>

</launch>
